src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +34,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
```

Comment:
You can consider moving the validation to the new classes that you have created.